src/utils/useMenuState.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect } from 'react';
+import { Router } from 'next/router';
+
+interface MenuStateReturn {
+  isMenuOpen: boolean;
+  openMenu: (event?: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  handleMenuClick: (
+    event?: React.MouseEvent<HTMLAnchorElement, MouseEvent>
+  ) => void;
+}
+
+/**
+ * Keep track of the menu state and handle the opening and closing.
+ * Returns controls to the state.
+ * @param defaultOpen
+ */
+export function useMenuState(defaultOpen = false): MenuStateReturn {
+  const [isMenuOpen, setMenu] = useState<boolean>(defaultOpen);
```

Comment:
Nitpick;

```suggestion
  const [isMenuOpen, setMenuOpen] = useState<boolean>(defaultOpen);
```

You're not really "setting a menu" here.